src/mlpack/methods/bayesian_ridge/bayesian_ridge.cpp

```diff
@@ -0,0 +1,299 @@
+
+      // Update beta.
+      temp = t - omega.t() * phi;
+      beta = (n - gamma) / dot(temp, temp);
+
+      // Comptute the stopping criterion.
+      deltaAlpha += alpha;
+      deltaBeta += beta;
+      crit = abs(deltaAlpha/alpha + deltaBeta/beta);
+      i++;
+    }
+  Timer::Stop("bayesian_ridge_regression");
+  return Rmse(data, responses);
+}
+
+void BayesianRidge::Predict(const arma::mat& points,
+                            arma::rowvec& predictions) const
+{
+  arma::mat X = points;
+
+  // Center and normalize the points before applying the model
+  X.each_col() -= data_offset;
+  X.each_col() /= data_scale;
+  predictions = omega.t() * X + responses_offset;
+}
+
+void BayesianRidge::Predict(const arma::colvec& point,
+                            double& prediction) const
+{
+  arma::mat point_mat = arma::conv_to<arma::mat>::from(point);
+  arma::rowvec prediction_vec(1);
+  Predict(point_mat, prediction_vec);
+  prediction = prediction_vec[0];
+}
+
+
+void BayesianRidge::Predict(const arma::colvec& point,
+                            double& prediction,
+                            double& std) const
+{
+  arma::mat point_mat = arma::conv_to<arma::mat>::from(point);
+  arma::rowvec prediction_vec(1);
+  arma::rowvec std_vec(1);
+  Predict(point_mat, prediction_vec, std_vec);
+  prediction = prediction_vec[0];
+  std = std_vec[0];
+}
+
+void BayesianRidge::Predict(const arma::mat& points,
+                            arma::rowvec& predictions,
+                            arma::rowvec& std) const
+{
+  arma::mat X = points;
+
+  // Center and normalize the points before applying the model.
+  X.each_col() -= data_offset;
+  X.each_col() /= data_scale;
+  predictions = omega.t() * X + responses_offset;
+
+  // Compute the standard deviation of each prediction.
+  std = arma::zeros<arma::rowvec>(X.n_cols);
+  arma::colvec phi(X.n_rows);
+  for (size_t i = 0; i < X.n_cols; i++)
+    {
+      phi = X.col(i);
+      std[i] = sqrt(Variance()
+               + dot(phi.t() * matCovariance, phi));
+    }
+}
+
+double BayesianRidge::Rmse(const arma::mat& data,
+                           const arma::rowvec& responses) const
+{
+  arma::rowvec predictions;
+  Predict(data, predictions);
+  return sqrt(mean(square(responses - predictions)));
+}
+
+void BayesianRidge::CenterNormalize(const arma::mat& data,
+                                    const arma::rowvec& responses,
+                                    bool fit_intercept,
+                                    bool normalize,
+                                    arma::mat& data_proc,
+                                    arma::rowvec& responses_proc,
+                                    arma::colvec& data_offset,
+                                    arma::colvec& data_scale,
+                                    double& responses_offset)
+{
+  // Initialize the offsets to their neutral forms.
+  data_offset = arma::zeros<arma::colvec>(data.n_rows);
```

Comment:
Please use camel casing for all names; see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more information. There are other places in the code as well, which should be adjusted.